festim/hydrogen_transport_problem.py

```diff
@@ -122,9 +134,21 @@ def define_markers_and_measures(self):
                 tags_facets.append(sub_dom.id)
             if isinstance(sub_dom, F.VolumeSubdomain1D):
                 # find all cells in subdomain and mark them as sub_dom.id
-                entities = sub_dom.locate_subdomain_entities(self.mesh.mesh, vdim)
+                self.volume_subdomains.append(sub_dom)
+                entities = sub_dom.locate_subdomain_entities(self.mesh, vdim)
                 tags_volumes[entities] = sub_dom.id
 
+        # check that subdomains are connected
+        if len(self.volume_subdomains) > 1:
+            all_borders = []
+            for vol in self.volume_subdomains:
+                for border in vol.borders:
+                    all_borders.append(border)
+            sorted_borders = np.sort(all_borders).reshape(int(len(all_borders) / 2), 2)
+            for i in range(0, len(sorted_borders) - 1):
+                if sorted_borders[i][1] != sorted_borders[i + 1][0]:
+                    raise ValueError("Subdomain borders don't match to each other ")
```

Comment:
I would have this in a separate method and check that the borders are within the mesh bounds at the same time!